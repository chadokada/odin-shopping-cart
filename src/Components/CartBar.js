import React, {useState} from 'react';
import '../Styles/App.css';
import shoppingCart from '../Images/Icons/shopping-cart.png';
import Cart from './Cart';

const CartBar = (
    {
      cart, 
      decrementItem, 
      incrementItem, 
      handleQuantityInput, 
      handleCheckOut
    }
  ) => {

  const getCartSize = () => {
    let cartSize = 0;
    for (const item in cart) {
      cartSize += cart[item].quantity;
    }
    return cartSize;
  }  
  
  const cartSize = getCartSize();

  let cartStatus = ''
  if (cartSize > 0) {
    cartStatus = ' full'
  };

  let [showCart, setShowCart] = useState(false);

  const toggleCart = () => {
    if (showCart === false) {
      setShowCart(true)
    } else {
      setShowCart(false)
    }
  }

  return (
  <div className="top-bar">
    <div className='store-name'>Butter and Sugar</div>

    <div className={`shopping-cart${cartStatus}`} onClick={toggleCart} data-testid='cart-button'>
      <div className='cart-size' data-testid='cart-size'>{cartSize}</div>
      <img alt="cart-icon" src={shoppingCart} className={`cart-icon${cartStatus}`}/>
    </div>
    <Cart 
      cart={cart} 
      showCart={showCart} 
      toggleCart={toggleCart}
      decrementItem={decrementItem}
      incrementItem={incrementItem}
      handleQuantityInput={handleQuantityInput}
      handleCheckOut={handleCheckOut}
    />
  </div>
  )
}

export default CartBar;