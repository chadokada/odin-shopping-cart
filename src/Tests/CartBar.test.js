import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CartBar from '../Components/CartBar';

describe('Test empty cart', () => {
  const onInputMock = jest.fn();
  const cart = {}
  
  it('should show no items in cart', () => {
    const { container } = render(
      <CartBar cart={cart} handleQuantityInput={onInputMock}/>
    );

    const cartSize = screen.getByTestId('cart-size');
  
    expect(cartSize.textContent).toMatch('0');
    expect(container).toMatchSnapshot();
  });
  
  it('should open cart side drawer', () => {
    const { container } = render(
      <CartBar cart={cart} handleQuantityInput={onInputMock}/>
    );
    
    const cartButton = screen.getByTestId('cart-button')
    userEvent.click(cartButton)
  
    expect(container).toMatchSnapshot()
  });
})

describe('Test cart with four items', () => {
  const cart = {
    'Citrus Bag' : {
      name: 'Citrus Bag', 
      price: 45.00,
      source: 'citrus_bag.png', 
      type:'bags',
      quantity: 1
    },
    'Bowler' : {
      name: 'Bowler', 
      price: 25.00, 
      source: 'bowler.jpg', 
      type:'hats',
      quantity: 3
    }
  };

  it('should show no items in cart', () => {
    const onInputMock = jest.fn();
    const { container } = render(
      <CartBar cart={cart} handleQuantityInput={onInputMock}/>
    );

    const cartSize = screen.getByTestId('cart-size');
  
    expect(cartSize.textContent).toMatch('4');
    expect(container).toMatchSnapshot();
  });


})
