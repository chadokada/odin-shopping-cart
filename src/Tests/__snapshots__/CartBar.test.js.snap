// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cart with four items should show no items in cart 1`] = `
<div>
  <div
    class="top-bar"
  >
    <div
      class="store-name"
    >
      Butter and Sugar
    </div>
    <div
      class="shopping-cart full"
      data-testid="cart-button"
    >
      <div
        class="cart-size"
        data-testid="cart-size"
      >
        4
      </div>
      <img
        alt="cart-icon"
        class="cart-icon full"
        src="shopping-cart.png"
      />
    </div>
    <div
      class="cart-drawer"
    >
      <div
        class="cart-overlay"
      />
      <div
        class="cart-container"
      >
        <div
          class="cart-header"
        >
          <h2>
            Your Shopping Cart
          </h2>
          <i
            class="fa-solid fa-x"
          />
        </div>
        <div
          class="cart-items-container"
        >
          <div
            class="cart-item"
            name="Citrus Bag"
            type="bags"
          >
            <div
              class="cart-item-img-container"
            >
              <img
                alt="[object Object]"
                class="cart-item-img"
                src="citrus_bag.png"
              />
            </div>
            <div
              class="cart-item-side"
            >
              <div
                class="cart-item-info"
              >
                <div
                  class="cart-item-name"
                >
                  Citrus Bag
                </div>
                <div
                  class="cart-item-price"
                >
                  $45.00
                </div>
              </div>
              <div
                class="cart-item-quantity-container"
              >
                <button>
                  -
                </button>
                <input
                  class="cart-item-quantity-input"
                  type="number"
                  value="1"
                />
                <button>
                  +
                </button>
              </div>
            </div>
          </div>
          <div
            class="cart-item"
            name="Bowler"
            type="hats"
          >
            <div
              class="cart-item-img-container"
            >
              <img
                alt="[object Object]"
                class="cart-item-img"
                src="bowler.jpg"
              />
            </div>
            <div
              class="cart-item-side"
            >
              <div
                class="cart-item-info"
              >
                <div
                  class="cart-item-name"
                >
                  Bowler
                </div>
                <div
                  class="cart-item-price"
                >
                  $25.00
                </div>
              </div>
              <div
                class="cart-item-quantity-container"
              >
                <button>
                  -
                </button>
                <input
                  class="cart-item-quantity-input"
                  type="number"
                  value="3"
                />
                <button>
                  +
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="cart-total"
        >
          Total: $
          120.00
        </div>
        <button
          class="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test empty cart should open cart side drawer 1`] = `
<div>
  <div
    class="top-bar"
  >
    <div
      class="store-name"
    >
      Butter and Sugar
    </div>
    <div
      class="shopping-cart"
      data-testid="cart-button"
    >
      <div
        class="cart-size"
        data-testid="cart-size"
      >
        0
      </div>
      <img
        alt="cart-icon"
        class="cart-icon"
        src="shopping-cart.png"
      />
    </div>
    <div
      class="cart-drawer open"
    >
      <div
        class="cart-overlay"
      />
      <div
        class="cart-container"
      >
        <div
          class="cart-header"
        >
          <h2>
            Your Shopping Cart
          </h2>
          <i
            class="fa-solid fa-x"
          />
        </div>
        <div
          class="cart-items-container"
        >
          <div>
            No items yet...
          </div>
        </div>
        <div
          class="cart-total"
        >
          Total: $
          0.00
        </div>
        <button
          class="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test empty cart should show no items in cart 1`] = `
<div>
  <div
    class="top-bar"
  >
    <div
      class="store-name"
    >
      Butter and Sugar
    </div>
    <div
      class="shopping-cart"
      data-testid="cart-button"
    >
      <div
        class="cart-size"
        data-testid="cart-size"
      >
        0
      </div>
      <img
        alt="cart-icon"
        class="cart-icon"
        src="shopping-cart.png"
      />
    </div>
    <div
      class="cart-drawer"
    >
      <div
        class="cart-overlay"
      />
      <div
        class="cart-container"
      >
        <div
          class="cart-header"
        >
          <h2>
            Your Shopping Cart
          </h2>
          <i
            class="fa-solid fa-x"
          />
        </div>
        <div
          class="cart-items-container"
        >
          <div>
            No items yet...
          </div>
        </div>
        <div
          class="cart-total"
        >
          Total: $
          0.00
        </div>
        <button
          class="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;
